standalone.tpl

```diff
@@ -0,0 +1,552 @@
+    """
+
+
+@cli.group(invoke_without_command=True)
+@click.option(
+    "--namespace", type=str, default="default", help="Kubernetes namespace to use"
+)
+@click.option(
+    "--taxonomy-repo-url",
+    type=str,
+    default=DEFAULT_REPO_URL,
+    help="URL of the taxonomy repository - for SDG only",
+)
+@click.option(
+    "--taxonomy-repo-branch",
+    type=str,
+    help="Branch of the taxonomy repository - for SDG only",
+)
+@click.option(
+    "--taxonomy-repo-pr",
+    type=str,
+    help="Pull request number of the taxonomy repository - for SDG only",
+)
+@click.option(
+    "--storage-class",
+    type=str,
+    default="standard",
+    help="Storage class to use for the PersistentVolumeClaim - for SDG only",
+)
+@click.option(
+    "--serving-endpoint",
+    type=str,
+    help="Serving endpoint for SDG - for SDG only",
+)
+@click.option(
+    "--serving-model",
+    type=str,
+    help="Serving model for SDG - for SDG only",
+)
+@click.option(
+    "--nproc-per-node",
+    type=str,
+    help="Number of processes per node - for training only",
+)
+@click.pass_context
+def run(
+    ctx: click.Context,
+    namespace: typing.Optional[str] = "default",
+    taxonomy_repo_url: str = "",
+    taxonomy_repo_branch: typing.Optional[str] = "",
+    taxonomy_repo_pr: typing.Optional[str] = "",
+    storage_class: typing.Optional[str] = "standard",
+    serving_endpoint: typing.Optional[str] = None,
+    serving_model: typing.Optional[str] = None,
+    nproc_per_node: typing.Optional[str] = 3,
+):
+    """
+    Execute the distributed training on Kubernetes.
+
+    Args:
+        namespace (str): The namespace to use for the setup process.
+        taxonomy_repo_url (str): The URL of the taxonomy repository. For SDG only.
+        taxonomy_repo_branch (str): The branch of the taxonomy repository. For SDG only.
+        taxonomy_repo_pr (int): The pull request number of the taxonomy repository. For SDG only.
+        storage_class (str): The storage class to use for the PersistentVolumeClaim. For SDG only.
+        serving_endpoint (str): The serving endpoint for SDG. For SDG only.
+        serving_model (str): The serving model for SDG. For SDG only.
+        nproc_per_node (str): The number of processes per node. For training only.
+
+    Returns:
+        None
+    """
+    ctx.ensure_object(dict)
+    ctx.obj["namespace"] = namespace
+    ctx.obj["taxonomy_repo_url"] = taxonomy_repo_url
+    ctx.obj["taxonomy_repo_branch"] = taxonomy_repo_branch
+    ctx.obj["taxonomy_repo_pr"] = taxonomy_repo_pr
+    ctx.obj["storage_class"] = storage_class
+    ctx.obj["serving_endpoint"] = serving_endpoint
+    ctx.obj["serving_model"] = serving_model
+    ctx.obj["nproc_per_node"] = nproc_per_node
+
+    # MAIN WORKFLOW SEQUENCE
+    # When the script is simply called like: 'python standalone.py run'
+    # We will run the entire workflow
+    if ctx.invoked_subcommand is None:
+        logger.info("Running setup for SDG.")
+        ctx.invoke(sdg)
```

Comment:
This throws the following error when running `python standalone.py run`

`TypeError: sdg() missing 1 required positional argument: 'ctx'`

Can be fixed by 
```suggestion
        ctx.invoke(sdg(ctx))
```
But this also looks quite wrong. Is there a better way to pass the ctx into the `sdg`?

Reply:
Sorry, I forgot the `@click.pass_context` decorator :). Now both:

```
python standalone.py run
```

And

```
python standalone.py run sdg
```

Should work :)